src/main/java/com/synopsys/integration/detect/workflow/blackduck/developer/aggregate/RapidScanResultAggregator.java

```diff
@@ -99,18 +118,70 @@ private List<RapidScanComponentDetail> aggregateComponentData(List<DeveloperScan
             addVulnerabilityData(resultView, vulnerabilityViolations, securityGroupDetail);
             addLicenseData(resultView, licenseViolations, licenseGroupDetail);
         }
+        
         return componentDetails;
     }
 
-    private RapidScanComponentDetail createDetail(DeveloperScansScanView view) {
-        String componentName = view.getComponentName();
-        String componentVersion = view.getVersionName();
+    private HashMap<String, List<String>> directToTransitiveChildren = new HashMap<>();
+    private HashMap<String, String[]> directUpgradeGuidanceVersions = new HashMap<>();
+    /**
+     * This is handled here because we have to use the data in a "look back" type of scenario.  
+     * We are getting lists of components and they in turn have transitive guidance if they are 
+     * transitive components.  We must base this list on the direct dependency component that come
+     * in the trans. guidance section, effectively reversing the order.  By handling that here, we
+     * can get a proper lookup of direct comp. -> child component(s).  Once this is completed,
+     * we can put together the string messages needed.
+     * 
+     * @param resultView 
+     * @param results
+     */
+    private void compileTransitiveGuidance(ScanFullResultView resultView, List<ScanFullResultView> results) {
+        List<ScanFullResultItemsTransitiveUpgradeGuidanceView> transitiveGuidance = resultView.getTransitiveUpgradeGuidance();
+        if (transitiveGuidance.size() <= 0) {
+            return;
+        }
+
+        String childExternalId = resultView.getExternalId();
+        if (resultView.getAllVulnerabilities().size() > 0) {// we'll only count things that HAVE vulnerabilities
```

Comment:
```suggestion
        if (transitiveGuidance.size() > 0 && resultView.getAllVulnerabilities().size() > 0) {
        // we'll only count things that HAVE vulnerabilities
        String childExternalId = resultView.getExternalId();
```

Reply:
This is a minor change (probably needs further refactor as I could not get the suggestion to work properly.
I'll approve.

Reply:
Quite easily done.... I'll be pushing in a minute.